src/raft/raft.rs

```diff
@@ -191,6 +192,9 @@ pub struct Raft<T: Storage> {
 
     /// tag is only used for logging
     tag: String,
+
+    /// Record the last instant of each peer's heartbeat response.
+    pub last_heartbeats: HashMap<u64, Instant>,
```

Comment:
I prefer keeping in peer too. Peer can add a method step to inspect the message too. Actually, receive any valid message from other peer should consider that peer is active.